pkg/apis/core/validation/backupbucket.go

```diff
@@ -47,7 +49,45 @@ func ValidateBackupBucketSpec(spec *core.BackupBucketSpec, fldPath *field.Path)
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("seedName"), spec.SeedName, "seed must not be empty"))
 	}
 
-	allErrs = append(allErrs, validateSecretReference(spec.SecretRef, fldPath.Child("secretRef"))...)
+	allErrs = append(allErrs, validateCredentials(spec, fldPath)...)
+
+	return allErrs
+}
+
+func validateCredentials(spec *core.BackupBucketSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// How to achieve backward compatibility between secretRef and credentialsRef?
+	// - if secretRef is set, credentialsRef must be set and refer the same secret
+	// - if secretRef is not set, then credentialsRef must refer a WorkloadIdentity
+	//
+	// After the sync in the strategy, we can have the following cases:
+	// - both secretRef and credentialsRef are unset, which we forbid here
+	// - both can be set but refer to different resources, which we forbid here
+	// - secretRef can be unset only when workloadIdentity is used, which we respect here
+
+	if spec.CredentialsRef == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("credentialsRef"), "must be set to refer a Secret or WorkloadIdentity"))
+	} else {
+		allErrs = append(allErrs, ValidateCredentialsRef(*spec.CredentialsRef, fldPath.Child("credentialsRef"))...)
+
+		if spec.CredentialsRef.GroupVersionKind().String() == corev1.SchemeGroupVersion.WithKind("Secret").String() {
+			if spec.SecretRef.Namespace != spec.CredentialsRef.Namespace || spec.SecretRef.Name != spec.CredentialsRef.Name {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("secretRef"), "must refer to the same secret as `spec.credentialsRef`"))
+			}
+		} else {
+			emptySecretRef := corev1.SecretReference{}
+			if spec.SecretRef != emptySecretRef {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("secretRef"), "must not be set when `spec.credentialsRef` refer to resource other than secret"))
+			}
+		}
+
+		// TODO(vpnachev): Allow WorkloadIdentities once the support in the controllers and components is fully implemented.
+		if spec.CredentialsRef.APIVersion == securityv1alpha1.SchemeGroupVersion.String() &&
+			spec.CredentialsRef.Kind == "WorkloadIdentity" {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("credentialsRef"), "support for workload identity as backup credentials is not yet fully implemented"))
```

Comment:
A general comment: I think we should stick to one of Secret or secret and WorkloadIdentity or workload identity in all validation messages
```suggestion
			allErrs = append(allErrs, field.Forbidden(fldPath.Child("credentialsRef"), "support for WorkloadIdentity as backup credentials is not yet fully implemented"))
```

Reply:
Done, I also looked around and aligned a few more places. Basically, when we are talking about the resource/kind I am using `WorkloadIdentity`, when we are talking about the feature/technology it is `workload identity`.